cylc/uiserver/app.py

```diff
@@ -542,15 +542,15 @@ def set_sub_server(self):
             auth=self.authobj,
         )
 
-    def set_auth(self):
+    def set_auth(self) -> Authorization:
         """Create authorization object.
         One for the lifetime of the UIServer.
         """
         return Authorization(
             getpass.getuser(),
-            self.config.CylcUIServer.user_authorization,
-            self.config.CylcUIServer.site_authorization,
-            self.log
+            self.config.CylcUIServer.user_authorization.to_dict(),
+            self.config.CylcUIServer.site_authorization.to_dict(),
```

Comment:
I have just pulled and now getting an error when accessing `cylc hub`

```
[W 2024-07-12 14:51:13.403 CylcHubApp manager:363] cylc.uiserver | extension failed loading with message: AttributeError("'dict' object has no attribute 'to_dict'")
    Traceback (most recent call last):
      File "~/.conda/envs/cylc8/lib/python3.11/site-packages/jupyter_server/extension/manager.py", line 359, in load_extension
        extension.load_all_points(self.serverapp)
      File "~/.conda/envs/cylc8/lib/python3.11/site-packages/jupyter_server/extension/manager.py", line 231, in load_all_points
        return [self.load_point(point_name, serverapp) for point_name in self.extension_points]
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "~/.conda/envs/cylc8/lib/python3.11/site-packages/jupyter_server/extension/manager.py", line 231, in <listcomp>
        return [self.load_point(point_name, serverapp) for point_name in self.extension_points]
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "~/.conda/envs/cylc8/lib/python3.11/site-packages/jupyter_server/extension/manager.py", line 222, in load_point
        return point.load(serverapp)
               ^^^^^^^^^^^^^^^^^^^^^
      File "~/.conda/envs/cylc8/lib/python3.11/site-packages/jupyter_server/extension/manager.py", line 150, in load
        return loader(serverapp)
               ^^^^^^^^^^^^^^^^^
      File "~/.conda/envs/cylc8/lib/python3.11/site-packages/jupyter_server/extension/application.py", line 474, in _load_jupyter_server_extension
        extension.initialize()
      File "~/.conda/envs/cylc8/lib/python3.11/site-packages/jupyter_server/extension/application.py", line 436, in initialize
        self._prepare_handlers()
      File "~/.conda/envs/cylc8/lib/python3.11/site-packages/jupyter_server/extension/application.py", line 326, in _prepare_handlers
        self.initialize_handlers()
      File "~/github/cylc-uiserver/cylc/uiserver/app.py", line 464, in initialize_handlers
        self.authobj = self.set_auth()
                       ^^^^^^^^^^^^^^^
      File "~/github/cylc-uiserver/cylc/uiserver/app.py", line 552, in set_auth
        self.config.CylcUIServer.site_authorization.to_dict(),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    AttributeError: 'dict' object has no attribute 'to_dict'
```

Reply:
I couldn't see a case where this would be a dict, no I know.

Easy fix.